### PR TITLE
feat(ci): update github actions to node 16

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #638 

## Description of the change:
This change updates GitHub actions node version from 12 that is deprecated to v16

## Motivation for the change:
Running workflows produces warning of the deprecated node 12 version

## How to manually test:
1. *Insert steps here...*
2. *...*

### CODE REVIEW

File: .github/workflows/test-ci-reusable.yml [hunk 1]
```diff
@@ -35,14 +35,14 @@ jobs:
...
+    - uses: actions/checkout@v4
       with:
         repository: ${{ inputs.repository }}
         ref: ${{ inputs.ref }}
-    - uses: actions/setup-go@v2
+    - uses: actions/setup-go@v4
       with:
         go-version: '1.20.*'
-    - uses: actions/cache@v2
+    - uses: actions/cache@v4
```
Comment (mwangggg) [inline .github/workflows/test-ci-reusable.yml:45]: pretty sure the latest version of actions/cache is v3

Comment (aali309) [reply .github/workflows/test-ci-reusable.yml:45]: yes: actions/setup-go and actions/checkout only are v4

File: .github/workflows/test-ci-command.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
         && (!contains(github.event.issue.labels.*.name, 'safe-to-test') || github.event.issue.user.name != github.event.comment.user.name)
       run: exit 1
     - name: React to comment
-      uses: actions/github-script@v4
+      uses: actions/github-script@v6
```
Comment (mwangggg) [inline .github/workflows/test-ci-command.yml:37]: v6/v5 has some breaking changes... I think they no longer have the `github.reactions.createForIssueComment` option

Comment (aali309) [reply .github/workflows/test-ci-command.yml:37]: yeah seen it.

**Comment by mwangggg** [comment]
[`actions/github-script@v4`](https://github.com/aali309/cryostat-operator/blob/001132c0e04c5e0138221aeb1a5a98770e5d2f64/.github/workflows/test-ci-command.yml#L58) still uses node12 so we want to update it to `v6`, just with the replacement for `github.reactions.createForIssueComment` and `github.pulls.get`

File: .github/workflows/test-ci-command.yml [hunk 3]
```diff
@@ -34,11 +34,11 @@ jobs:
...
       with:
         script: |
             const {owner, repo} = context.issue
-            github.reactions.createForIssueComment({
+            github.rest.reactions.createForIssueComment({github.reactions.createForIssueComment
```
Comment (tthvo) [inline .github/workflows/test-ci-command.yml:41]: remove `github.reactions.createForIssueComment`?

**Review by mwangggg — APPROVED** [review]
looks good to me :+1:

**Review by ebaron — APPROVED** [review]

**Comment by ebaron** [comment]
Hey @aali309, would you be able to rebase? I tried GitHub's automated rebase, but it seems to have made the commits unverified.

**Comment by andrewazores** [comment]
> Hey @aali309, would you be able to rebase? I tried GitHub's automated rebase, but it seems to have made the commits unverified.

https://github.com/orgs/community/discussions/11639

**Comment by aali309** [comment]
> Hey @aali309, would you be able to rebase? I tried GitHub's automated rebase, but it seems to have made the commits unverified.

done

**Pull request merged**